numpy/core/tests/test_multiarray.py

```diff
@@ -9493,6 +9500,104 @@ class MyAlwaysEqualNew(MyAlwaysEqual):
         assert_equal(array != my_always_equal, 'ne')
 
 
+@pytest.mark.parametrize("op", [operator.eq, operator.ne])
+@pytest.mark.parametrize(["dt1", "dt2"], [
+        ([("f", "i")], [("f", "i")]),  # structured comparison (successfull)
+        ("M8", "d"),  # impossible comparison: result is all True or False
+        ("d", "d"),  # valid comparison
+        ])
+def test_equal_subclass_no_override(op, dt1, dt2):
+    # Test how the three different possible code-paths deal with subclasses
+
+    class MyArr(np.ndarray):
+        called_wrap = 0
+
+        def __array_wrap__(self, new):
+            type(self).called_wrap += 1
+            return super().__array_wrap__(new)
+
+    numpy_arr = np.zeros(5, dtype=dt1)
+    my_arr = np.zeros(5, dtype=dt2).view(MyArr)
+
+    assert type(op(numpy_arr, my_arr)) is MyArr
+    assert type(op(my_arr, numpy_arr)) is MyArr
+    # We expect 2 calls (more if there were more fields):
+    assert MyArr.called_wrap == 2
+
+
+@pytest.mark.parametrize(["dt1", "dt2"], [
+        ("M8[ns]", "d"),
+        # Note: timedelta currently promotes (and always did) :(
+        #       so it does not line in here.
```

Comment:
I thought you are fixing that too?  (also spelling: "line"?)  Or promotes to int?

Reply:
I really really want to pull the plug in this as well!  But here I only changed datetime and timedelta, not timedelta and integer promotions (I think we should keep the two changesets distinct)